### ISSUE TITLE
Feature Request: `not` schema

### ISSUE THREAD

**Issue by alturkovic** [ghandic/jsf#105]
Support `not` schema: https://json-schema.org/understanding-json-schema/reference/combining#not

**Comment by alturkovic** [comment]
For an example, generate an odd number between 10-20:

```json
{
  "type": "integer",
  "minimum": 10,
  "allOf": [
    {
      "maximum": 20
    }
  ],
  "not": {
    "allOf": [
      {
        "multipleOf": 2
      }
    ]
  }
}
```

The example provided is deliberately more complex than it needs to be to show the composing nature of the problem.

**Comment by ghandic** [comment]
Do you have a proposed solution for this?

**Comment by alturkovic** [comment]
https://github.com/json-schema-faker/json-schema-faker/blob/d4403ae6cdba2206fe86399900c4095de8db7d2a/src/lib/core/utils.mjs#L350
https://github.com/json-schema-faker/json-schema-faker/blob/d4403ae6cdba2206fe86399900c4095de8db7d2a/src/lib/core/traverse.mjs#L64

It seems like the original author built a new schema and "manually" inversed some of the properties and then replaced the original schema... 
For an example, it does not handle `multipleOf` that I mentioned originally.

Another approach might be to collect all the constraints imposed on a certain value and then generate a value matching all those constraints?
For an example:
- `type` must be `integer`
- value must be in range 10 .. 20
- must not be divisible by 2

But I assume that could get very complicated very fast with lots of constraints...

**Comment by alturkovic** [comment]
I think that the problem is much bigger than I initially assumed: https://github.com/ghandic/jsf/blob/8db07552459670d13ca6f1e37c6479771882d802/jsf/parser.py#L255

The current implementation will only take `type` into consideration if it encounters that keyword, so valid definitions such as the one below will not work:
```json
{
  "type": "integer",
  "allOf": [
    {
      "minimum": 10
    },
    {
      "maximum": 20
    }
  ]
}
```

The above should be equivalent to:

```json
{
  "type": "integer",
  "minimum": 10,
  "maximum": 20
}
```

I think the `__parse_definition` needs to do some more sofisticated "merging" than it currently does. Unfortunately, I do not know Python to jump in with any PRs...

**Comment by ghandic** [comment]
I would have thought the above wouldn't be valid without a type in the all of objects.

**Comment by alturkovic** [comment]
Yup, that was my assumption also at first, but things are a bit more complicated...
I have tested the schema above with the original JSF and [here](https://www.jsonschemavalidator.net/). Seems to be valid.
